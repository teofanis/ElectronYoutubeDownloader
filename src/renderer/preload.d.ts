import { IpcRequest, PromisedIpcResponse } from 'interfaces';

declare global {
  interface Window {
    electron: {
      store: {
        getState: <T>() => T;
        setState: <T>(
          recipe: (draft: T) => void,
          description?: string | undefined
        ) => T;
        subscribe: <T>(
          listener: (state: T, description?: string | undefined) => void
        ) => () => void;
      };
      ipc: {
        send: <T>(
          channel: string,
          request: IpcRequest
        ) => PromisedIpcResponse<T>;
      };
    };
  }
}

export {};
