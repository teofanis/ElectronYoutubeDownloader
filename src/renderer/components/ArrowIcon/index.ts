// eslint-disable-next-line import/prefer-default-export
export { default as ArrowIcon } from './ArrowIcon';
