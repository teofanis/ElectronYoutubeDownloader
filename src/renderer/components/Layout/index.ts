/* eslint-disable import/prefer-default-export */
export { default as Layout } from './Layout';
