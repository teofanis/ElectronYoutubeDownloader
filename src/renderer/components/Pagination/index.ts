// eslint-disable-next-line import/prefer-default-export
export { default as Pagination } from './Pagination';
