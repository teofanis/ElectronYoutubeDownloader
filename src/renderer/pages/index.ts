import { Home } from './Home';

// eslint-disable-next-line import/prefer-default-export
export { Home };
