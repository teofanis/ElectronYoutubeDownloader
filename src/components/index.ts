// eslint-disable-next-line import/prefer-default-export
import { Button } from 'components/Button';
import { CancelButton } from 'components/CancelButton';
import { DownloadableItem } from 'components/DownloadableItem';
import { DownloadableItemControls } from 'components/DownloadableItemControls';
import { DownloadableItemTransition } from 'components/DownloadableItemTransition';
import { DownloadableItemTransitionContainer } from 'components/DownloadableItemTransitionContainer';
import { DownloadButton } from 'components/DownloadButton';
import { Downloader } from 'components/Downloader';
import { DownloadItemControl } from 'components/DownloadItemControl';
import { FileInput } from 'components/FileInput';
import { Heading } from 'components/Heading';
import { InputError } from 'components/InputError';
import { Layout } from 'components/Layout';
import { Logo } from 'components/Logo';
import { ProgressBar } from 'components/ProgressBar';
import { TextInput } from 'components/TextInput';
import { DownloadableItemTransition } from 'components/DownloadableItemTransition';
// Automatic import of components
export {
  Layout,
  Heading,
  Logo,
  Downloader,
  TextInput,
  FileInput,
  Button,
  DownloadButton,
  InputError,
  ProgressBar,
  CancelButton,
  DownloadableItem,
  DownloadableItemControls,
  DownloadItemControl,
  DownloadableItemTransitionContainer,
  DownloadableItemTransition,
  DownloadableItemTransition,
  // Automatic export of components
};
