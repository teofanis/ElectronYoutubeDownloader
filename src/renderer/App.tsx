import { Home } from 'pages';
import { MemoryRouter as Router, Route, Routes } from 'react-router-dom';
import './global.css';

export default function App() {
  return (
    <Router>
      <Routes>
        <Route path="/" element={<Home />} />
      </Routes>
    </Router>
  );
}
