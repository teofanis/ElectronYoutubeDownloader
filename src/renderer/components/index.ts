// eslint-disable-next-line import/prefer-default-export
import { Button } from 'renderer/components/Button';
import { CancelButton } from 'renderer/components/CancelButton';
import { DownloadableItem } from 'renderer/components/DownloadableItem';
import { DownloadableItemControls } from 'renderer/components/DownloadableItemControls';
import { DownloadableItemTransition } from 'renderer/components/DownloadableItemTransition';
import { DownloadableItemTransitionContainer } from 'renderer/components/DownloadableItemTransitionContainer';
import { DownloadButton } from 'renderer/components/DownloadButton';
import { Downloader } from 'renderer/components/Downloader';
import { DownloaderControls } from 'renderer/components/DownloaderControls';
import { DownloadItemControl } from 'renderer/components/DownloadItemControl';
import { FileInput } from 'renderer/components/FileInput';
import { Heading } from 'renderer/components/Heading';
import { InputError } from 'renderer/components/InputError';
import { Layout } from 'renderer/components/Layout';
import { Logo } from 'renderer/components/Logo';
import { ProgressBar } from 'renderer/components/ProgressBar';
import { TextInput } from 'renderer/components/TextInput';
import { Pagination } from 'renderer/components/Pagination';
// Automatic import of components
export {
  Layout,
  Heading,
  Logo,
  Downloader,
  TextInput,
  FileInput,
  Button,
  DownloadButton,
  InputError,
  ProgressBar,
  CancelButton,
  DownloadableItem,
  DownloadableItemControls,
  DownloadItemControl,
  DownloadableItemTransitionContainer,
  DownloadableItemTransition,
  DownloaderControls,
  Pagination,
  // Automatic export of components
};
