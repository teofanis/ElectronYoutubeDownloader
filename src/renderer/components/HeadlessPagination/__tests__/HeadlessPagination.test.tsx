import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { IPaginationProps } from 'interfaces';
import { HeadlessPagination } from 'renderer/components';

const setupPagination = ({
  currentPage = 5,
  setCurrentPage = () => {},
  className = '',
  truncableText = '...',
  truncableClassName = '',
  totalPages = 10,
  edgePageCount = 1,
  middlePagesSiblingCount = 2,
  children = (
    <>
      <HeadlessPagination.PrevButton dataTestId="prev">
        Previous
      </HeadlessPagination.PrevButton>

      <div className="flex items-center justify-center flex-grow">
        <HeadlessPagination.PageButton
          dataTestIdActive="active"
          dataTestIdInactive="inactive"
        />
      </div>

      <HeadlessPagination.NextButton dataTestId="next">
        Next
      </HeadlessPagination.NextButton>
    </>
  ),
}: Partial<IPaginationProps>) =>
  render(
    <HeadlessPagination
      currentPage={currentPage}
      setCurrentPage={setCurrentPage}
      className={className}
      truncableText={truncableText}
      truncableClassName={truncableClassName}
      totalPages={totalPages}
      edgePageCount={edgePageCount}
      middlePagesSiblingCount={middlePagesSiblingCount}
    >
      {children}
    </HeadlessPagination>
  );

describe('HeadlessPagination', () => {
  it('renders correctly with basic setup of prevButton, pageButton and nextButton', () => {
    const { asFragment } = setupPagination({});

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with different truncable text', () => {
    const { asFragment } = setupPagination({
      truncableText: '---',
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with advanced setup of prevButton, pageButton and nextButton using Tailwind', () => {
    const { asFragment } = setupPagination({
      className: 'flex items-center w-full h-10 text-sm select-none',
      truncableClassName: 'w-10 px-0.5 text-center',
      children: (
        <>
          <HeadlessPagination.PrevButton className="flex items-center mr-2 text-gray-500 cursor-pointer hover:text-gray-600 dark:hover:text-gray-200 focus:outline-none">
            Previous
          </HeadlessPagination.PrevButton>

          <div className="flex items-center justify-center flex-grow">
            <HeadlessPagination.PageButton
              activeClassName="bg-primary-50 dark:bg-opacity-0 text-primary-600 dark:text-white"
              inactiveClassName="text-gray-500"
              className="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer"
            />
          </div>

          <HeadlessPagination.NextButton className="flex items-center mr-2 text-gray-500 opacity-50 hover:text-gray-600 dark:hover:text-gray-200 focus:outline-none">
            Next
          </HeadlessPagination.NextButton>
        </>
      ),
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('clicking on previous calls mockSetCurrentPage with currentPage - 1', () => {
    const mockSetCurrentPage = jest.fn();

    setupPagination({
      currentPage: 4,
      totalPages: 5,
      setCurrentPage: mockSetCurrentPage,
    });

    expect(screen.getByTestId('active-page-button').textContent).toEqual('5');

    screen.getByTestId('prev').click();

    expect(mockSetCurrentPage).toHaveBeenCalledWith(3);
  });

  it('clicking on next calls mockSetCurrentPage with currentPage + 1', () => {
    const mockSetCurrentPage = jest.fn();

    setupPagination({
      currentPage: 1,
      totalPages: 5,
      setCurrentPage: mockSetCurrentPage,
    });

    expect(screen.getByTestId('active-page-button').textContent).toEqual('2');

    screen.getByTestId('next').click();

    expect(mockSetCurrentPage).toHaveBeenCalledWith(2);
  });

  it('clicking on current page calls mockSetCurrentPage with currentPage', () => {
    const mockSetCurrentPage = jest.fn();

    setupPagination({
      currentPage: 5,
      totalPages: 10,
      setCurrentPage: mockSetCurrentPage,
    });

    expect(screen.getByTestId('active-page-button').textContent).toEqual('6');

    screen.getByTestId('inactive-page-button-10').click();

    expect(mockSetCurrentPage).toHaveBeenCalledWith(9);
  });
});
