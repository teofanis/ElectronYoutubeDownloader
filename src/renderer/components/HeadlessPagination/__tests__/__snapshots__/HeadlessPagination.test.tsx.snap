// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadlessPagination renders correctly with advanced setup of prevButton, pageButton and nextButton using Tailwind 1`] = `
<DocumentFragment>
  <div
    class="flex items-center w-full h-10 text-sm select-none"
  >
    <button
      class="flex items-center mr-2 text-gray-500 cursor-pointer hover:text-gray-600 dark:hover:text-gray-200 focus:outline-none"
      type="button"
    >
      Previous
    </button>
    <div
      class="flex items-center justify-center flex-grow"
    >
      <span
        class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer text-gray-500"
      >
        1
      </span>
      <span
        class="w-10 px-0.5 text-center"
      >
        ...
      </span>
      <span
        class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer text-gray-500"
      >
        4
      </span>
      <span
        class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer text-gray-500"
      >
        5
      </span>
      <span
        class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer bg-primary-50 dark:bg-opacity-0 text-primary-600 dark:text-white"
      >
        6
      </span>
      <span
        class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer text-gray-500"
      >
        7
      </span>
      <span
        class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer text-gray-500"
      >
        8
      </span>
      <span
        class="w-10 px-0.5 text-center"
      >
        ...
      </span>
      <span
        class="flex items-center justify-center h-10 w-10 rounded-full cursor-pointer text-gray-500"
      >
        10
      </span>
    </div>
    <button
      class="flex items-center mr-2 text-gray-500 opacity-50 hover:text-gray-600 dark:hover:text-gray-200 focus:outline-none"
      type="button"
    >
      Next
    </button>
  </div>
</DocumentFragment>
`;

exports[`HeadlessPagination renders correctly with basic setup of prevButton, pageButton and nextButton 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <button
      data-testid="prev"
      type="button"
    >
      Previous
    </button>
    <div
      class="flex items-center justify-center flex-grow"
    >
      <span
        data-testid="inactive-page-button-1"
      >
        1
      </span>
      <span>
        ...
      </span>
      <span
        data-testid="inactive-page-button-4"
      >
        4
      </span>
      <span
        data-testid="inactive-page-button-5"
      >
        5
      </span>
      <span
        data-testid="active-page-button"
      >
        6
      </span>
      <span
        data-testid="inactive-page-button-7"
      >
        7
      </span>
      <span
        data-testid="inactive-page-button-8"
      >
        8
      </span>
      <span>
        ...
      </span>
      <span
        data-testid="inactive-page-button-10"
      >
        10
      </span>
    </div>
    <button
      data-testid="next"
      type="button"
    >
      Next
    </button>
  </div>
</DocumentFragment>
`;

exports[`HeadlessPagination renders correctly with different truncable text 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <button
      data-testid="prev"
      type="button"
    >
      Previous
    </button>
    <div
      class="flex items-center justify-center flex-grow"
    >
      <span
        data-testid="inactive-page-button-1"
      >
        1
      </span>
      <span>
        ---
      </span>
      <span
        data-testid="inactive-page-button-4"
      >
        4
      </span>
      <span
        data-testid="inactive-page-button-5"
      >
        5
      </span>
      <span
        data-testid="active-page-button"
      >
        6
      </span>
      <span
        data-testid="inactive-page-button-7"
      >
        7
      </span>
      <span
        data-testid="inactive-page-button-8"
      >
        8
      </span>
      <span>
        ---
      </span>
      <span
        data-testid="inactive-page-button-10"
      >
        10
      </span>
    </div>
    <button
      data-testid="next"
      type="button"
    >
      Next
    </button>
  </div>
</DocumentFragment>
`;
